### PR TITLE
Sample: Use default bin and obj directories locally

### PR DESCRIPTION
**Pull request by bricelam**
Changing `BaseIntermediateOutputPath` prevents "dotnet ef" from working.

Resolves aspnet/EntityFrameworkCore#12220

cc @richlander @divega @ajcvickers

### CODE REVIEW

**Comment by ajcvickers** [comment]
@richlander Can this get merged? Is there someone else who should review?

**Comment by MichaelSimons** [comment]
Is everyone satisfied with the guidance in the readmes (https://github.com/dotnet/dotnet-docker/blob/master/samples/dotnetapp/dotnet-docker-dev-in-container.md#requirements & https://github.com/dotnet/dotnet-docker/blob/master/samples/aspnetapp/aspnet-docker-dev-in-container.md#requirements)?  The guidance is rather open-ended.  If there are problems/drawbacks with changing the local bin/obj, perhaps the wording should be adjusted to specifically call out you should only change the container bin/obj path.

> To avoid conflicts between container usage and your local environment, you need to use a different set of obj and bin folders for each environment.
> 
> Make this change with the following steps:
> 
> Delete your existing obj and bin folders manually or use dotnet clean.
> Add a Directory.Build.props file to your project to use a different set of obj and bin folders for your local and container environments.

**Comment by ajcvickers** [comment]
@richlander @MichaelSimons Any update? Can this get merged?

**Comment by MichaelSimons** [comment]
@ajcvickers - Do you have thoughts on the question I posed (https://github.com/dotnet/dotnet-docker/pull/1033#issuecomment-480901576)?  I am of the opinion the readme should be updated as well per this change.

**Comment by ajcvickers** [comment]
@MichaelSimons I have no opinion on that; who would be able to make a call as to whether this is needed or not?

**Comment by raix** [comment]
@ajcvickers has anyone tested if a "local" build will overwrite the "container" folder? - the two builds need to coexist without interfering...

**Comment by divega** [comment]
> If there are problems/drawbacks with changing the local bin/obj, perhaps the wording should be adjusted to specifically call out you should only change the container bin/obj path.

@MichaelSimons yes, I think it is a good idea to reflect this change in the guidance as well. Are you asking that @bricelam makes this changes also in the PR or can you do it after this after it is merged?

**Comment by ajcvickers** [comment]
@bricelam Can you answer the question from @raix

**Comment by MichaelSimons** [comment]
The guidance and code should be in sync otherwise the samples are not coherant.  I think it could be as simple as making the following tweak to the two readmes ((https://github.com/dotnet/dotnet-docker/blob/master/samples/dotnetapp/dotnet-docker-dev-in-container.md#requirements & https://github.com/dotnet/dotnet-docker/blob/master/samples/aspnetapp/aspnet-docker-dev-in-container.md#requirements)).

```
To avoid conflicts between container usage and your local environment, you need to use a different set of `obj` and `bin` folders for your container environment.

 Make this change with the following steps:

 1. Delete your existing obj and bin folders manually or use `dotnet clean`.
 2. Add a [Directory.Build.props](Directory.Build.props) file to your project to use a different set of `obj` and `bin` folders for your container environments.
```

**Comment by bricelam** [comment]
> Will a "local" build will overwrite the "container" folder?

It won't. The output assemblies will be like this. There won't be any overlap.

* bin/Debug/netcoreapp2.2/Sample.dll
* bin/**container**/Debug/netcoreapp2.2/Sample.dll
* obj/Debug/netcoreapp2.2/Sample.dll
* obj/**container**/Debug/netcoreapp2.2/Sample.dll

File: samples/dotnetapp/dotnet-docker-dev-in-container.md [hunk 1]
```diff
@@ -20,12 +20,12 @@ You can also [download the repository as a zip](https://github.com/dotnet/dotnet
 
 This approach relies on [volume mounting](https://docs.docker.com/engine/admin/volumes/volumes/) (that's the `-v` argument in the following commands) to mount source into the container (without using a Dockerfile). You may need to [Enable shared drives (Windows)](https://docs.docker.com/docker-for-windows/#shared-drives) or [file sharing (macOS)](https://docs.docker.com/docker-for-mac/#file-sharing) first.
 
-To avoid conflicts between container usage and your local environment, you need to use a different set of `obj` and `bin` folders for each environment.
+To avoid conflicts between container usage and your local environment, you need to use a different set of `obj` and `bin` folders for your container environment.
```
Comment (MichaelSimons) [inline samples/dotnetapp/dotnet-docker-dev-in-container.md]: This change should get mirrored to [aspnet](https://github.com/dotnet/dotnet-docker/blob/master/samples/aspnetapp/aspnet-docker-dev-in-container.md) as well.

Comment (bricelam) [reply samples/dotnetapp/dotnet-docker-dev-in-container.md]: The wording was different there, and still made sense to me the way it was...

**Pull request merged**